### DISCUSSION TITLE
Tuya TS011F_plug_3 detected as TS011F_plug_1 (probably)

### DISCUSSION THREAD

**Discussion by Pathen85** [Koenkk/zigbee2mqtt#16075] [Devices]
Hey,
so I bought the [Tuya Plug with energy monitoring capabilities](https://www.zigbee2mqtt.io/devices/TS011F_plug_1.html). When I connect it to Z2M it gets immediately picked up as TS011F_plug_1

![2](https://user-images.githubusercontent.com/12153262/211529222-5ca87801-c9ee-40dd-94b0-4d936eb0de4a.PNG)

According to the Z2M website, this device works by reporting its energy status in a set interval. However, for me it just reported Null for the power, current, voltage and energy values. On/Off works fine. So what I did was opening the Dev Console and poll the mentioned values from the device.

![1](https://user-images.githubusercontent.com/12153262/211529899-08056779-ed75-450e-904c-3d49b14bab51.PNG)

As you can see, the plug correctly measures those paremeters and responds to my poll. After that those values show up under 'Details', however they won't refresh on their own.

![3](https://user-images.githubusercontent.com/12153262/211530244-51f87c3c-04e3-49fc-97d7-c42f5e5e88ad.PNG)

As you can see in the screenshot, the plug is turned off but still shows the same values. So what I assume happened here, is that I got the plug with the newer firmware that doesn't suppot automatic reporting, but it was somehow still detected as plug_1. Is there any way to tell Z2M to handle it as plug_3? Or am I able to set up automatic polling for this plug?

Thanks in advance!
Henrik

**Comment by Koenkk** [comment]
Can you provide the data/database.db entry of this plug?

**Comment by Pathen85** [comment]
Of course! Thanks for looking into this!

```
{
  "id": 28,
  "type": "Router",
  "ieeeAddr": "0x70b3d52b6001bc58",
  "nwkAddr": 14003,
  "manufId": 4660,
  "manufName": "_TZ3000_okaz9tjs",
  "powerSource": "Mains (single phase)",
  "modelId": "TS011F",
  "epList": [
    1
  ],
  "endpoints": {
    "1": {
      "profId": 260,
      "epId": 1,
      "devId": 81,
      "inClusterList": [
        0,
        6,
        3,
        4,
        5,
        57345,
        2820,
        1794
      ],
      "outClusterList": [],
      "clusters": {
        "genBasic": {
          "attributes": {
            "manufacturerName": "_TZ3000_okaz9tjs",
            "zclVersion": 2,
            "appVersion": 100,
            "modelId": "TS011F",
            "powerSource": 1
          }
        },
        "haElectricalMeasurement": {
          "attributes": {
            "activePower": 3,
            "rmsVoltage": 227,
            "rmsCurrent": 23
          }
        },
        "genOnOff": {
          "attributes": {
            "onOff": 1
          }
        },
        "seMetering": {
          "attributes": {
            "currentSummDelivered": [
              0,
              1
            ]
          }
        }
      },
      "binds": [
        {
          "cluster": 6,
          "type": "endpoint",
          "deviceIeeeAddress": "0x00124b0029dedea9",
          "endpointID": 1
        },
        {
          "cluster": 2820,
          "type": "endpoint",
          "deviceIeeeAddress": "0x00124b0029dedea9",
          "endpointID": 1
        },
        {
          "cluster": 1794,
          "type": "endpoint",
          "deviceIeeeAddress": "0x00124b0029dedea9",
          "endpointID": 1
        }
      ],
      "configuredReportings": [],
      "meta": {}
    }
  },
  "appVersion": 100,
  "stackVersion": 1,
  "hwVersion": 3,
  "dateCode": "20210625",
  "swBuildId": "20B+TZSKT11BS108",
  "zclVersion": 2,
  "interviewCompleted": true,
  "meta": {},
  "lastSeen": 1673468096862,
  "defaultSendRequestWhen": "immediate"
}
```

**Reply by Koenkk** [reply]
This has already been fixed in the dev branch and the fix will be include in the 1 February release.  (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)

**Comment by Pathen85** [comment]
Thanks for the info and I appreciate you taking your time to look into this!

**Reply by Pathen85** [reply]
I switched to the dev branch and it works now.
